mhw-cs-plugin-loader/Preloader.cpp

```diff
@@ -26,16 +27,13 @@ SafetyHookInline g_mh_main_ctor_hook{};
 CoreClr* s_coreclr = nullptr;
 NativePluginFramework* s_framework = nullptr;
 
-// TODO(Andoryuuta): AOB scan for these.
-namespace preloader::address {
-    const uint64_t IMAGE_BASE = 0x140000000;
-    const uint64_t PROCESS_SECURITY_COOKIE = IMAGE_BASE + 0x4bf4be8;
-    const uint64_t SECURITY_COOKIE_INIT_GETTIME_RET = IMAGE_BASE + 0x27422e2;
-    const uint64_t SCRT_COMMON_MAIN_SEH = IMAGE_BASE + 0x27414f4;
-    const uint64_t WINMAIN = IMAGE_BASE + 0x13a4c00;
-
-}  // namespace preloader::address
-
+// The default value that MSVC uses for the IMAGE_LOAD_CONFIG_DIRECTORY64.SecurityCookie.
+const uint64_t MSVC_DEFAULT_SECURITY_COOKIE_VALUE = 0x2B992DDFA232L;
+const Pattern SCRT_COMMON_MAIN_PATTERN = Pattern::from_string("48 89 5C 24 08 57 48 83 EC 30 B9 01 00 00 00 E8 ?? ?? ?? ?? 84 C0");
+const Pattern WINMAIN_CALL_PATTERN = Pattern::from_string("E8 ?? ?? ?? ?? 0F B7 D8 E8 ?? ?? ?? ?? 4C 8B C0 44 8B CB 33 D2 48 8D ?? ?? ?? ?? ??");
+const int64_t WINMAIN_CALL_PATTERN_OFFSET = 28;
+const Pattern MHMAIN_CALL_PATTERN = Pattern::from_string("BA 00 00 08 00 48 8B CF E8 ?? ?? ?? ?? 4C 89 3F C7 87 ?? ?? ?? ?? FF FF FF FF");
```

Comment:
Maybe we should think about pulling these from the AddressRepository as well. If not, there should at least be some central file to hold patterns. (Including stuff like the main update function, etc)

Reply:
Yeah, I'm not sure how much we want to duplicate in both the native and managed code - we basically "solved" all the same isuses in the managed code already. For these addresses, it would be nice to pull them from the address repo, scan in parallel, and cached them so that the game startup time.

Should I just lift that entire bit in to the native code and turn the managed `AddressRecords` class into a proxy to the native code?

Reply:
True. Yeah I think keeping everything in native code should be fine. It could get a bit annoying with the plugin cache though. You might wanna leave that one on the managed side to not overcomplicate things.

Reply:
Okay, the address repo has been moved to native. Should be good for a final review now :)